### PR TITLE
New: [NA] [Style] [Header] Created a Header section [ED-10522]

### PR DESCRIPTION
**Pull request by TzviRabinovitch**

### CODE REVIEW

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 1]
```diff
@@ -441,7 +444,141 @@ private function add_content_style_section() {
...
+		$this->add_control(
+			'heading_icon_style_title',
+			[
+				'type' => Controls_Manager::HEADING,
+				'label' => esc_html__( 'Icon', 'elementor' ),
+
+			]
+		);
+
+		$this->add_responsive_control( 'icon_size', [
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:493]: Same here for example.

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 2]
```diff
@@ -11,8 +11,16 @@
...
+  	--n-accordion-title-normal-color: #{$editor-darker};
+	--n-accordion-title-hover-color: #{$editor-darker};
+	--n-accordion-title-active-color: #{$editor-darker};
+	--n-accordion-icon-normal-color: var(--n-accordion-title-normal-color);
+  	--n-accordion-icon-hover-color: var(--n-accordion-title-hover-color);
+	--n-accordion-icon-active-color: var(--n-accordion-title-active-color);
+	--n-accordion-icon-gap: 0 15px;
 
-	.e-n-accordion {
+  .e-n-accordion {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:23]: Indendentation.

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 3]
```diff
@@ -34,27 +42,45 @@
...
+			  	gap: var( --n-accordion-icon-gap );
+			  	color: var(--n-accordion-title-normal-color);
+			  	align-items: center;
 
-				&-text h1,h2,h3,h4,h5,h6,p {
-					margin-block: 0;
+			  	&-header{
+				  	display: flex;
+
+					h1,h2,h3,h4,h5,h6,p {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:52]: Spaces in between.

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 4]
```diff
@@ -63,32 +89,68 @@
...
+								}
+							}
+						}
+					}
 				}
 
-				.e-closed {
-					display: flex;
-				}
+				.e-n-accordion-item-title-icon{
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:138]: Space

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 5]
```diff
@@ -63,32 +89,68 @@
...
+					.e-opened {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:139]: Empty line

File: tests/playwright/enums/display-states.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export const displayState = {
```
Comment (hein-obox) [inline tests/playwright/enums/display-states.js:1]: Nice

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 7]
```diff
@@ -441,7 +452,148 @@ private function add_content_style_section() {
...
+			]
+		);
+
+		$this->add_control( $state . '_' . $context . '_color', [
+			'label' => esc_html__( 'Color', 'elementor' ),
+			'type' => Controls_Manager::COLOR,
+			'selectors' => [
+				'{{WRAPPER}}' => $variable . ': {{VALUE}};',
+			],
+		] );
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:591]:
```suggestion
		$this->add_control(
	                $state . '_' . $context . '_color',
	                [
			                'label' => esc_html__( 'Color', 'elementor' ),
			                'type' => Controls_Manager::COLOR,
			                'selectors' => [
				                '{{WRAPPER}}' => $variable . ': {{VALUE}};',
			                ],
	                ]
                );
```

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 8]
```diff
@@ -175,27 +183,29 @@ protected function register_controls() {
...
 		);
 
-		$this->add_responsive_control( 'accordion_item_title_icon_position', [
-			'label' => esc_html__( 'Position', 'elementor' ),
-			'type' => Controls_Manager::CHOOSE,
-			'options' => [
-				'start' => [
-					'title' => esc_html__( 'Start', 'elementor' ),
-					'icon' => 'eicon-h-align-left',
+		$this->add_responsive_control( 'accordion_item_title_icon_position',
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:186]: `accordion_item_title_icon_position` should be on the next line

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 9]
```diff
@@ -114,57 +118,61 @@ protected function register_controls() {
...
-			'separator' => 'before',
-			'options' => [
-				'start' => [
-					'title' => esc_html__( 'Start', 'elementor' ),
-					'icon' => 'eicon-flex eicon-align-start-h',
-				],
-				'center' => [
-					'title' => esc_html__( 'Center', 'elementor' ),
-					'icon' => 'eicon-h-align-center',
+		$this->add_responsive_control( 'accordion_item_title_position_horizontal',
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:142]: `accordion_item_title_position_horizontal` should be on the next line

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 10]
```diff
@@ -114,57 +118,61 @@ protected function register_controls() {
...
-			'default' => [
-				[
-					'item_title' => esc_html__( 'Item #1', 'elementor' ),
-				],
-				[
-					'item_title' => esc_html__( 'Item #2', 'elementor' ),
-				],
-				[
-					'item_title' => esc_html__( 'Item #3', 'elementor' ),
+		$this->add_control( 'items',
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:121]: `items` should be on the next line

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 11]
```diff
@@ -89,16 +91,18 @@ protected function register_controls() {
...
-			'label' => esc_html__( 'Title', 'elementor' ),
-			'type' => Controls_Manager::TEXT,
-			'default' => esc_html__( 'Item Title', 'elementor' ),
-			'placeholder' => esc_html__( 'Item Title', 'elementor' ),
-			'label_block' => true,
-			'dynamic' => [
-				'active' => true,
-			],
-		] );
+		$repeater->add_control( 'item_title',
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:94]: `item_title` - next line

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/nested-accordion/assets/scss/frontend.scss (hunk 3 above)
Comment (hein-obox) [reply modules/nested-accordion/assets/scss/frontend.scss:51]: Not changed online.

File: modules/nested-accordion/assets/scss/frontend.scss (hunk 4 above)
Comment (hein-obox) [reply modules/nested-accordion/assets/scss/frontend.scss:136]:
```suggestion
				.e-n-accordion-item-title-icon {
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 12]
```diff
@@ -11,8 +11,16 @@
...
+  	--n-accordion-title-normal-color: #{$editor-darker};
+	--n-accordion-title-hover-color: #{$editor-darker};
+	--n-accordion-title-active-color: #{$editor-darker};
+	--n-accordion-icon-normal-color: var(--n-accordion-title-normal-color);
+  	--n-accordion-icon-hover-color: var(--n-accordion-title-hover-color);
+	--n-accordion-icon-active-color: var(--n-accordion-title-active-color);
+	--n-accordion-icon-gap: 0 15px;
 
-	.e-n-accordion {
+.e-n-accordion {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:23]: @Barcov 
Tabs are missing.
I should be like how it was before.

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 13]
```diff
@@ -34,27 +42,44 @@
...
-					margin-block: 0;
+			  	&-header{
+				  	display: flex;
+
+					h1, h2, h3, h4, h5, h6, p {
+						margin-block: 0;
+					}
+				}
+
+				&-text{
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:57]:
```suggestion
				&-text {
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 14]
```diff
@@ -34,27 +42,44 @@
 				justify-content: var(--n-accordion-title-justify-content);
 				border-radius: var(--n-accordion-border-radius);
 				flex-grow: var(--n-menu-title-flex-grow);
+			  	gap: var( --n-accordion-icon-gap );
+			  	color: var(--n-accordion-title-normal-color);
+			  	align-items: center;
 
-				&-text h1,h2,h3,h4,h5,h6,p {
-					margin-block: 0;
+			  	&-header{
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:49]:
```suggestion
			  	&-header {
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 15]
```diff
@@ -63,32 +88,69 @@
...
+						}
+					}
 				}
 			}
 
 			&:not([open]) {
 
-				.e-opened {
-					display: none;
+			  	&:hover{
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:118]:
```suggestion
			  	&:hover {
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 16]
```diff
@@ -63,32 +88,69 @@
...
+				  	.e-n-accordion-item-title {
+							color : var(--n-accordion-title-hover-color);
+
+							&-icon {
+								span{
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:123]:
```suggestion
								span {
```

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 17]
```diff
@@ -63,32 +88,69 @@
...
-
-				.e-closed {
-					display: none;
+					color: var(--n-accordion-title-active-color);
+
+					&-icon {
+						.e-opened {
+							display: flex;
+					  	}
+					  	.e-closed {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:99]: empty line

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 18]
```diff
@@ -63,32 +88,69 @@
...
 			&:not([open]) {
 
-				.e-opened {
-					display: none;
+			  	&:hover {
+				  	.e-n-accordion-item-title {
+							color : var(--n-accordion-title-hover-color);
+
+							&-icon {
+								span {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:123]: empty line

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 19]
```diff
@@ -63,32 +88,69 @@
 			}
 
 			&[open] {
-
 			  	.e-n-accordion-item-title {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:91]: empty line above

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 20]
```diff
@@ -63,32 +88,69 @@
...
+					&-icon {
+						.e-opened {
+							display: flex;
+					  	}
+					  	.e-closed {
+							display: none;
+					  	}
+
+					  	span {
+							& > i {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:104]: empty line

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 21]
```diff
@@ -63,32 +88,69 @@
...
+					}
 				}
 			}
 
 			&:not([open]) {
 
-				.e-opened {
-					display: none;
+			  	&:hover {
+				  	.e-n-accordion-item-title {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:119]: empty line

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 22]
```diff
@@ -456,7 +471,150 @@ private function add_content_style_section() {
...
+		);
+
+		$this->add_control(
+			'heading_header_style_title',
+			[
+				'type' => Controls_Manager::HEADING,
+				'label' => esc_html__( 'Title', 'elementor' ),
+
+			]
+		);
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:490]: @Barcov Add empty line here

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 23]
```diff
@@ -456,7 +471,150 @@ private function add_content_style_section() {
...
+		$this->add_group_control( Group_Control_Typography::get_type(),
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:491]: @Barcov Group_Control_Typography::get_type() - next line

File: tests/playwright/sanity/modules/nested-accordion/nested-accordion.test.js [hunk 24]
```diff
@@ -635,7 +693,7 @@ async function getIcon( nestedAccordionItem, iconIndex ) {
  * Set Nested Accordion Title Tag (H1-H6,div,span,p)
  *
  * @param {string} optionToSelect          - value of select option i.e. h1,h2,h3,h4,h5,h6,div,span,p
- * @param {string} nestedAccordionWidgetId -- id of the nested accordion widget
+ * @param {string} nestedAccordionWidgetId - - id of the nested accordion widget
```
Comment (louiswol94) [inline tests/playwright/sanity/modules/nested-accordion/nested-accordion.test.js:696]: @Barcov Why this change?

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 25]
```diff
@@ -456,7 +471,150 @@ private function add_content_style_section() {
...
+		$this->end_controls_section();
+
+	}
+
+	private function add_header_color_style( $state, $context ) {
+
+		$translated_tab_text = esc_html__( 'Normal', 'elementor' );
+
+		$variable = '--n-accordion-' . $context . '-' . $state . '-color';
+		switch ( $state ) {
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:589]: Add empty lone above `switch`

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 26]
```diff
@@ -456,7 +471,150 @@ private function add_content_style_section() {
...
+		$this->start_controls_tabs( 'header_icon_color_style' );
+
+		foreach ( [ 'normal', 'hover', 'active' ] as $state ) {
+			$this->add_header_color_style( $state, 'icon' );
+		}
+
+		$this->end_controls_tabs();
+
+		$this->end_controls_section();
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:581]: @Barcov Remove this empty line

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 27]
```diff
@@ -456,7 +471,150 @@ private function add_content_style_section() {
...
+		$this->add_control(
+			'heading_icon_style_title',
+			[
+				'type' => Controls_Manager::HEADING,
+				'label' => esc_html__( 'Icon', 'elementor' ),
+
+			]
+		);
+
+		$this->add_responsive_control( 'icon_size',
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:522]: @Barcov `icon_size` - next line

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 28]
```diff
@@ -63,32 +88,69 @@
...
-				.e-opened {
-					display: flex;
-				}
-
-				.e-closed {
-					display: none;
+					color: var(--n-accordion-title-active-color);
+
+					&-icon {
+						.e-opened {
```
[the remark this answers is not in the document]
Comment (louiswol94) [reply modules/nested-accordion/assets/scss/frontend.scss:96]: @Barcov We normally add empty lines between the selectors.

File: modules/nested-accordion/assets/scss/frontend.scss (hunk 19 above)
Comment (louiswol94) [reply modules/nested-accordion/assets/scss/frontend.scss:91]: @Barcov Meaning this empty line should be here (it should not be removed).

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 29]
```diff
@@ -2,8 +2,10 @@
 namespace Elementor\Modules\NestedAccordion\Widgets;
 
 use Elementor\Controls_Manager;
+use Elementor\Core\Kits\Documents\Tabs\Global_Typography;
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:5]: @Barcov Where is `Global_Typography` used?

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 30]
```diff
@@ -456,7 +471,150 @@ private function add_content_style_section() {
...
+				break;
+		}
+		$this->start_controls_tab(
+			'header_' . $state . '_' . $context,
+			[
+				'label' => $translated_tab_text,
+			]
+		);
+
+		$this->add_control( $state . '_' . $context . '_color',
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:604]: @Barcov `$state . '_' . $context . '_color'` - next line

File: modules/nested-accordion/assets/scss/frontend.scss (hunk 21 above)
Comment (hein-obox) [reply modules/nested-accordion/assets/scss/frontend.scss:122]: There should be an empty line between the hover and the item title selector.

**Review by hein-obox — DISMISSED** [review]

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 31]
```diff
@@ -1,99 +1,166 @@
...
+                                fill: var(--n-accordion-icon-active-color);
+                            }
+                        }
+                    }
+                }
+            }
+
+            &:not([open]) {
+
+                &:hover {
```
Comment (hein-obox) [inline modules/nested-accordion/assets/scss/frontend.scss:121]: Empty line missing between `&:hover {` and  `.e-n-accordion-item-title {`

File: modules/nested-accordion/assets/scss/frontend.scss [hunk 32]
```diff
@@ -1,99 +1,166 @@
...
-					}
-				}
-
-				& > span {
-					cursor: pointer;
-				}
-			}
-		}
-	}
+    --n-accordion-title-font-size: 20px;
```
Comment (louiswol94) [inline modules/nested-accordion/assets/scss/frontend.scss:2]: @Barcov @hein-obox Something's mot right here. The file looks like it has 2 spaces instead of 4 for indentation, so this is a big diff.

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 33]
```diff
@@ -456,7 +469,155 @@ private function add_content_style_section() {
...
+				'selectors' => [
+					'{{WRAPPER}}' => $variable . ': {{VALUE}};',
+				],
+			]
+		);
+
+		$this->end_controls_tab();
+	}
+
+		/**
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:620]: Spacing issue with `/**`

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 34]
```diff
@@ -456,7 +469,155 @@ private function add_content_style_section() {
...
+
+		$this->start_controls_tabs( 'header_icon_color_style' );
+
+		foreach ( [ 'normal', 'hover', 'active' ] as $state ) {
+			$this->add_header_color_style( $state, 'icon' );
+		}
+
+		$this->end_controls_tabs();
+		$this->end_controls_section();
+
```
Comment (louiswol94) [inline modules/nested-accordion/widgets/nested-accordion.php:581]: @Barcov Delete this empty line.

**Review by louiswol94 — APPROVED** [review]
